sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -691,7 +691,9 @@ case class Pivot(
     pivotColumn: Expression,
     pivotValues: Seq[Literal],
     aggregates: Seq[Expression],
-    child: LogicalPlan) extends UnaryNode {
+    child: LogicalPlan,
+    groupByExprsImplicit: Boolean = false) extends UnaryNode {
+  override lazy val resolved = false // Pivot will be replaced after being resolved.
```

Comment:
Could you add a negative test case to show which errors we could issue when Pivot can't be resolved? If the error does not make sense to the end users, we can improve the function `checkAnalysis` in the trait `CheckAnalysis`